rclpy/src/rclpy/_rclpy.c

```diff
@@ -126,11 +135,12 @@ rclpy_trigger_guard_condition(PyObject * Py_UNUSED(self), PyObject * args)
   if (!PyArg_ParseTuple(args, "O", &pygc)) {
     return NULL;
   }
-  if (!PyCapsule_CheckExact(pygc)) {
-    Py_RETURN_FALSE;
-  }
 
-  rcl_guard_condition_t * gc = (rcl_guard_condition_t *)PyCapsule_GetPointer(pygc, NULL);
+  rcl_guard_condition_t * gc = (rcl_guard_condition_t *)PyCapsule_GetPointer(
+    pygc, "rcl_guard_condition_t");
+  if (!gc) {
+    return NULL;
```

Comment:
What is the expectation here? It should only return `NULL` when an error is set, right? But `PyCapsule_GetPointer` shouldn't set an error. Should a specific error message be set or should this return `Py_RETURN_NONE` instead (which would be a silent failure, probably not?)?

Same below for various other pointers.

Reply:
`PyCapsule_GetPointer` will raise an exception if the Capsule is not a valid capsule with a matching name. I think the assumption here is that the only reason for `gc` to be `NULL` is if `PyCapsule_GetPointer` fails and thus an exception is already being set so we need to return `NULL`.

Reply:
You are right, it [does](https://docs.python.org/3.5/c-api/capsule.html#c.PyCapsule_GetPointer) set an error when returning `NULL`. Sorry for the noise - should have checked the docs before.